### DISCUSSION TITLE
kong deploy with postgresql。how can i record response body

### DISCUSSION THREAD

**Discussion by zerotocode398** [Kong/kong#12416] [Help]
kong version is 2.5.0. i use admin api create file-log plugin 
![image](https://github.com/Kong/kong/assets/31309558/41eb2b49-eeab-4240-8014-d82fda73b8f9)

config.custom_fields_by_lua.req_body="return ngx.var.request_body" expect record request body
config.custom_fields_by_lua.rsp_body="return ngx.var.response_body" expect record response body
but only record request body。
**how can i record response body**
![image](https://github.com/Kong/kong/assets/31309558/8fdc9391-d1fc-4d0c-bddf-469c606b9d12)
![image](https://github.com/Kong/kong/assets/31309558/10a57b22-b5fd-4f6e-a14c-b134830f20d3)

**Comment by ms2008** [comment]
Here's one option offered https://discuss.konghq.com/t/logging-request-body-response-body-with-http-log/1205/17

**Reply by zerotocode398** [reply]
thanks your help.
but this doc used dbless mode. i think /config endpoint only support dbless mode.
i try demo. like
![image](https://github.com/Kong/kong/assets/31309558/c1c10134-e21f-44c3-a2d2-38fa1f0ee5e8)

**Reply by ms2008** [reply]
He's just providing an idea that you can totally extend to the traditional mode. Try this:

```
curl -X POST http://localhost:8001/plugins \
    --data "name=file-log"  \
    --data "config.path=/tmp/file.log"
    
curl -X POST http://localhost:8001/plugins \
    --data "name=pre-function"  \
    --data 'config.access=kong.log.set_serialize_value("request.body", kong.request.get_raw_body())' \
    --data 'config.body_filter=kong.log.set_serialize_value("response.body", kong.response.get_raw_body())'
```

**Reply by zerotocode398** [reply]
if i only set file-log. like
![image](https://github.com/Kong/kong/assets/31309558/2e523b04-6e03-480d-b643-31a9795b322f)
then i request
![image](https://github.com/Kong/kong/assets/31309558/f9dd1566-eaab-4eec-899b-8d3e220435e4)
![image](https://github.com/Kong/kong/assets/31309558/91564427-7a22-48a5-883b-ea22363f7601)

when i set both
![image](https://github.com/Kong/kong/assets/31309558/81bff21a-a2d6-4f32-a0df-c4af0a5a704b)
then i request
![image](https://github.com/Kong/kong/assets/31309558/bda1b29f-6496-47a2-9533-da4376ef5493)
![image](https://github.com/Kong/kong/assets/31309558/df7182dc-5de8-4e86-b427-0b8d303d935c)

not response body & client is error

**Reply by ms2008** [reply]
It looks like your upstream is not returning a valid response, and I was able to see the response body using the following steps:
```
curl -i -X  POST http://localhost:8001/services/ \
   --data "name=plugin-testing" \
   --data "url=http://httpbin.org/headers"

curl -i -X  POST http://localhost:8001/services/plugin-testing/routes \
   --data "paths[]=/test"

curl -X POST http://localhost:8001/plugins \
    --data "name=file-log"  \
    --data "config.path=/tmp/file.log"

curl -X POST http://localhost:8001/plugins \
    --data "name=pre-function"  \
    --data 'config.access=kong.log.set_serialize_value("request.body", kong.request.get_raw_body())' \
    --data 'config.body_filter=kong.log.set_serialize_value("response.body", kong.response.get_raw_body())'

curl -is http://localhost:8000/test
```

**Reply by zerotocode398** [reply]
What makes an effective response. must have head or type?
my demo return default http code 200 & body json.
